test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/BatchTests.cs

```diff
@@ -65,6 +65,51 @@ public async Task GivenAValidBundle_WhenSubmittingABatch_ThenSuccessIsReturnedFo
             ValidateOperationOutcome(resource.Entry[9].Response.Status, resource.Entry[9].Response.Outcome as OperationOutcome, _statusCodeMap[HttpStatusCode.NotFound], "Resource type 'Patient' with id '12334' couldn't be found.", IssueType.NotFound);
         }
 
+        [Fact]
+        [HttpIntegrationFixtureArgumentSets(dataStores: DataStore.CosmosDb)]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenAValidBundle_WhenSubmittingABatchTwiceWithAndWithoutChanges_ThenVersionIsCreatedWhenDataIsChanged()
+        {
+            var requestBundle = Samples.GetDefaultBatch().ToPoco<Bundle>();
+            using FhirResponse<Bundle> fhirResponse = await _client.PostBundleAsync(requestBundle);
+            Assert.NotNull(fhirResponse);
+            Assert.Equal(HttpStatusCode.OK, fhirResponse.StatusCode);
+
+            Bundle resource = fhirResponse.Resource;
+
+            Assert.Equal("201", resource.Entry[0].Response.Status);
+            Assert.Equal("201", resource.Entry[1].Response.Status);
+            Assert.Equal("201", resource.Entry[2].Response.Status);
+            Assert.Equal("1", resource.Entry[2].Resource.VersionId);
+            Assert.Equal("201", resource.Entry[3].Response.Status);
+            Assert.Equal(((int)Constants.IfMatchFailureStatus).ToString(), resource.Entry[4].Response.Status);
+            Assert.Equal("204", resource.Entry[5].Response.Status);
+            Assert.Equal("204", resource.Entry[6].Response.Status);
+            ValidateOperationOutcome(resource.Entry[7].Response.Status, resource.Entry[7].Response.Outcome as OperationOutcome, _statusCodeMap[HttpStatusCode.NotFound], "The route for \"/ValueSet/$lookup\" was not found.", IssueType.NotFound);
+            Assert.Equal("200", resource.Entry[8].Response.Status);
+            ValidateOperationOutcome(resource.Entry[9].Response.Status, resource.Entry[9].Response.Outcome as OperationOutcome, _statusCodeMap[HttpStatusCode.NotFound], "Resource type 'Patient' with id '12334' couldn't be found.", IssueType.NotFound);
+
+            // WhenSubmittingABatchTwiceWithNoDataChange_ThenServerShouldNotCreateAVersionSecondTimeAndSendOk
+            using FhirResponse<Bundle> fhirResponseAfterPostingSameBundle = await _client.PostBundleAsync(requestBundle);
+            Assert.NotNull(fhirResponseAfterPostingSameBundle);
+            Assert.Equal(HttpStatusCode.OK, fhirResponseAfterPostingSameBundle.StatusCode);
+
+            Bundle resourceAfterPostingSameBundle = fhirResponseAfterPostingSameBundle.Resource;
+
+            Assert.Equal("201", resourceAfterPostingSameBundle.Entry[0].Response.Status);
+            Assert.Equal("200", resourceAfterPostingSameBundle.Entry[1].Response.Status);
```

Comment:
Should entry 1's version id also be checked? It looks like it also wasn't changed.

Reply:
Entry 1, is a conditional create and next time there is a single match returning 200 ok with no resource information